rocketchat_API/APISections/chat.py

```diff
@@ -5,21 +5,56 @@
 class RocketChatChat(RocketChatBase):
     def chat_post_message(self, text, room_id=None, channel=None, **kwargs):
         """Posts a new chat message."""
-        if room_id:
-            if text:
-                return self.call_api_post(
-                    "chat.postMessage", roomId=room_id, text=text, kwargs=kwargs
-                )
-            return self.call_api_post("chat.postMessage", roomId=room_id, kwargs=kwargs)
+        if not (channel or room_id):
+            raise RocketMissingParamException("roomId or channel required")
+
+        text = "" if text is None else text
+
+        payload = {"text": text, **kwargs}
         if channel:
-            if text:
-                return self.call_api_post(
-                    "chat.postMessage", channel=channel, text=text, kwargs=kwargs
-                )
-            return self.call_api_post(
-                "chat.postMessage", channel=channel, kwargs=kwargs
-            )
-        raise RocketMissingParamException("roomId or channel required")
+            payload["channel"] = channel
+        if room_id:
+            payload["roomId"] = room_id
```

Comment:
If both channel and roomId are not specified, this method should rise an exception

`raise RocketMissingParamException("roomId or channel required")`